### DISCUSSION TITLE
Drum Editor horizontal scroll bar and Ctrl button fixed.

### DISCUSSION THREAD

**Discussion by terminator356** [muse-sequencer/muse#1157] [Team Posts]
Hey, give it a try. I fixed it up.
The Drum Editor Ctrl button now follows the canvas width.
The h-scroll is now the same size as the canvas.
Try out the extreme ranges of the splitters and observe that the Ctrl button
 does not overlap and 'gives some leeway' when at or near the left or right edge extremes.

**Comment by kybos** [comment]
Oh yes, that looks much better and cleaner, thank you!

BTW: I looked at the possibility to replace the Ctrl-related button texts with icons, as you had suggested in the PR, but I'm afraid the buttons are currently much too small for that, the icons would look bad at this size. But perhaps now with your new layout it would be possible to make even the Ctrl button higher, independent of the scrollbar? I will check the code...

**Comment by terminator356** [comment]
Yeah I was reminded of the ugly fact that those buttons (and several other things) are fixed size.
It affected how I had to make the new automatic layout class (we have one in the Arranger as well).
When you use fixed size, calling sizeHint() or minimumSizeHint() is no help, you have to 
 ask for the actual size(). That was a bummer.
I contemplated removing the fixed limits and letting the components down there
 grow to their natural size. But then we run into really ugly style variation issues.

I still believe we could make reasonable icons. I'll try a few things today...
One severe limitation with actual buttons is the gross style margins they impose.
That's why I made my Icon Button class, which has no button face or limiting borders.
(You used it in the Arranger strip 'show' button.)
Nothing but a good ol' full icon right up to the borders.
Although I really must get around to drawing a thin border around them ;-)

**Comment by terminator356** [comment]
> It affected how I had to make the new automatic layout class

Some musings about that layout class:
I put that class in a separate module so that in the future the PianoRoll Editor might use it as well.
In the future the piano graphic may become variable size. Why?
I've had some mild plans on the back-burner to allow showing note names - even in the piano.
Yes, non-percussion melodic note names and even groups of note names
 **are** available from fluidsynth, and now that MidiName support is here,
 that is a possibility particularly with all the LV2 synths.

**Comment by kybos** [comment]
> Yeah I was reminded of the ugly fact that those buttons (and several other things) are fixed size.

As I understand it, it's not even the fixed size which is the problem here. The height of the Ctrl button is currently bound to the hscroll height, which is generally too small for a sensible button (and which can get still smaller because it depends on the theme).

**Comment by terminator356** [comment]
Correct. Which means that if we were to remove fixed sizes,
 the height of the scrollbar would be dictated by the button height! That'd be much too high I think.

**Comment by kybos** [comment]
Exactly. To manipulate the height of the scrollbar is a no-go, that would break the design (I tried and it of course looked terribly ugly). So the only way is to make the button independent of the scrollbar somehow? Put it somewhere else, toolbar, or even the controller view together with the other two buttons, even if it's duplicated then...

**Comment by terminator356** [comment]
Actually I kind of like that it's directly underneath the controller panel now ;-)

**Comment by terminator356** [comment]
I will **freeze** my commits for now so Robert can proceed
 with confidence this weekend if he wants.

**Comment by kybos** [comment]
> Actually I kind of like that it's directly underneath the controller panel now ;-)

I agree that this is exactly the right place for it, there is only the problem with the size...
But I guess it's no big issue, it looks pretty acceptable now.

**Comment by kybos** [comment]
> Actually I kind of like that it's directly underneath the controller panel now ;-)
> 
> I agree that this is exactly the right place for it, there is only the problem with the size...
> But I guess it's no big issue, it looks pretty acceptable now.

I did a test install on KDE once again and there the Ctrl button really looks OK, almost the same height as the S and X buttons above. But e.g. on Cinnamon the scrollbar is smaller so the button is also too small. This unpredictable theme dependency is certainly not good (but also not critical, I guess).
But I noticed another thing: The drum list is now very small (in width) after a fresh install (same as the piano in piano roll).
